gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsService.java

```diff
@@ -0,0 +1,508 @@
+   */
+  enum ZoneField {
+    CREATION_TIME("creationTime"),
+    DESCRIPTION("description"),
+    DNS_NAME("dnsName"),
+    ZONE_ID("id"),
+    NAME("name"),
+    NAME_SERVER_SET("nameServerSet"),
+    NAME_SERVERS("nameServers");
+
+    private final String selector;
+
+    ZoneField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(ZoneField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(ZONE_ID.selector());
+      for (ZoneField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The fields of a DNS record.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@code DnsService#listDnsRecords(BigInteger, DnsRecordOptions...)} or {@code
+   * DnsService#listDnsRecords(String, DnsRecordOptions...)}. The name is always returned even if
+   * not selected.
+   */
+  enum DnsRecordField {
+    DNS_RECORDS("rrdatas"),
+    NAME("name"),
+    TTL("ttl"),
+    TYPE("type");
+
+    private final String selector;
+
+    DnsRecordField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(DnsRecordField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(NAME.selector());
+      for (DnsRecordField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The fields of a change request.
+   *
+   * <p>These values can be used to specify the fields to include in a partial response when calling
+   * {@code DnsService#applyChangeRequest(ChangeRequest, BigInteger, ChangeRequestFieldOptions...)}
+   * or {@code DnsService#applyChangeRequest(ChangeRequest, String, ChangeRequestFieldOptions...)}
+   * The ID is always returned even if not selected.
+   */
+  enum ChangeRequestField {
+    ID("id"),
+    START_TIME("startTime"),
+    STATUS("status"),
+    ADDITIONS("additions"),
+    DELETIONS("deletions");
+
+    private final String selector;
+
+    ChangeRequestField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(ChangeRequestField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(ID.selector());
+      for (ChangeRequestField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The sorting keys for listing change requests. The only currently supported sorting key is the
+   * change sequence.
```

Comment:
Per our discussion. I think change sequence is unclear.
Also, API seems to be strange and we should discuss it with the service owners.
We could possibly add it later (when there is more than one option). If we decided to take it out, 
then we should have an issue for adding it later.